pkg/platform/runtime/executors/executors.go

```diff
@@ -55,15 +55,16 @@ func (es *Executors) ExecutorSrc() (string, error) {
 func (es *Executors) Apply(sockPath string, targeter Targeter, env map[string]string, exes envdef.ExecutablePaths) error {
 	logging.Debug("Creating executors at %s, exes: %v", es.executorPath, exes)
 
-	// We need to cover the use case of someone running perl.exe/python.exe
-	// Proper fix scheduled here https://www.pivotaltracker.com/story/show/177845386
-	if rt.GOOS == "windows" {
-		for _, exe := range exes {
-			if !strings.HasSuffix(exe, exeutils.Extension) {
-				continue
+	executors := make(map[string]string) // map[alias]dest
+	for _, dest := range exes {
+		alias := filepath.Base(dest)
+		if rt.GOOS == "windows" {
+			ext := filepath.Ext(alias)
+			if ext != "" && ext != exeutils.Extension { // for non-.exe executables like pip.bat
+				alias = strings.TrimSuffix(alias, ext) + exeutils.Extension // setup alias pip.exe -> pip.bat
 			}
-			exes = append(exes, exe+exeutils.Extension) // Double up on the ext so only the first on gets dropped
 		}
+		executors[alias] = dest
```

Comment:
This is starting to become a bit hard to read, can you clean it up a bit? Maybe split it out into a function that clearly denotes what it's doing.